### PR TITLE
Backend Restruct, Frontend Tweaks and move to Postgres

### PR DESCRIPTION
**Pull request by 0xk1f0**
#### Basic Requirements

- [x] I have added Documentation https://github.com/CTF-Citadel/ctf-citadel.github.io/pull/33
- [x] I have tested my code
- [x] I have synced my branch with main

#### Describe your PR, what does it fix/add?

This unifies backend API-Endpoints to `/api/v1` which currently provide

- `/admin` - privileged actions
- `/user` - normal user actions
- `/account/auth` - authenticate an existing user
- `/account/new` - new account addition
- `/token/email` - email verify token generation
- `/token/password` - password reset token generation
- `/reset/[id].ts` - password reset validation

We also now enforce password strength and Username/Email validity, in client side code, as well as before db action is taken.

Also tweaks frontend design elements so we have some color unification.
Flowbite Components will need a unified look in general, which needs to be implemented on a `class` override basis, depending on the component used.

Finally, this PR migrates from MariaDB to Postgres, via a few DrizzleORM and Lucia restructs.

#### Is there anything you want to mention? (unchecked code, possible bugs, found problems, breaking compatibility, etc.)

\-

#### Is it ready for merging, or does it need work?

\-

### CODE REVIEW

File: src/lib/schema.ts [hunk 1]
```diff
@@ -1,158 +1,96 @@
...
+    event_id: text('event_id')
+        .references(() => events.id),
+    challenge_name: text('challenge_name').unique(),
+    challenge_category: text('challenge_category'),
+    challenge_difficulty: text('challenge_difficulty'),
+    challenge_description: text('challenge_description'),
+    container_file: text('container_file'),
+    static_file_url: text('static_file_url'),
+    needs_container: boolean('needs_container')
 });
```
Comment (Merrixx) [inline src/lib/schema.ts:64]: Add hints property to challenges

File: src/lib/schema.ts [hunk 2]
```diff
@@ -1,158 +1,96 @@
...
-		length: 256
-	}),
+export const teams = pgTable('teams', {
+    id: text('id').primaryKey(),
+    team_creator: text('team_creator').references(() => users.id),
+    team_name: text('team_name').unique(),
+    team_join_token: text('team_join_token'),
+    team_country_code: text('team_country_code'),
+    team_description: text('team_description')
 });
```
Comment (Merrixx) [inline src/lib/schema.ts:73]: Add team_points for buying hints

File: src/lib/schema.ts (hunk 1 above)
Comment (0xk1f0) [reply src/lib/schema.ts:64]: HInts might depend on Deployment API changes, not adding yet but noted.

File: src/lib/schema.ts (hunk 2 above)
Comment (0xk1f0) [reply src/lib/schema.ts:74]: added in https://github.com/CTF-Citadel/webapp/pull/92/commits/2a88c90dfb62a87faa4e4f214372d5c799a607cf

**Review by Merrixx — APPROVED** [review]
Checked

**Pull request merged**